### PR TITLE
Prepare API docs for v3.4.0

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Checklist

- [n/a] Has pytest style unit tests (and `pytest` passes).
- [n/a] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: doc/api/next_api_changes/deprecations/18003-AL.rst [hunk 1]
```diff
@@ -1,3 +0,0 @@
-mathtext.Fonts.destroy
```
Comment (QuLogic) [inline doc/api/next_api_changes/deprecations/18003-AL.rst:1]: All of `mathtext.Fonts` is deprecated, so this was dropped.

File: doc/api/next_api_changes/deprecations/18004-AL.rst [hunk 2]
```diff
@@ -1,4 +0,0 @@
-*facename* parameter of ``mathtext.Fonts.render_glyph``
```
Comment (QuLogic) [inline doc/api/next_api_changes/deprecations/18004-AL.rst:1]: All of `mathtext.Fonts` is deprecated, so this was dropped.

File: doc/api/next_api_changes/deprecations/18673-AL.rst [hunk 3]
```diff
@@ -1,3 +0,0 @@
-``ParasiteAxesAuxTransBase.update_viewlim``
```
Comment (QuLogic) [inline doc/api/next_api_changes/deprecations/18673-AL.rst:1]: All of `ParasiteAxesAuxTransBase` is deprecated, so this was dropped.

**Review by timhoffm — APPROVED** [review]
Looks good, but I didn't check every detail. I'll leave this open in case somebody else still wants to have a look. But you may proceed based on my review if nobody else shows up.

**Pull request merged**